### ISSUE TITLE
[3.7.27.2]: Nav tag doesn't output as expected when parent entry is disabled

### ISSUE THREAD

**Issue by GaryReckard** [craftcms/cms#10925]
### Description

When using the `{% nav %}` tag, as described [here](https://craftcms.com/docs/3.x/dev/tags.html#nav),  it seems to be incorrectly assigning the children of a disabled entry to the previously enabled entry.

### Steps to reproduce

1. Have a Structure, and have two root level entries "Parent 1" and "Parent 2", and two child entries "Child 1" and "Child 2" under "Parent 2". All entries are 'Enabled'.
2. Use the `{% nav %}` loop referenced above to output those entries in a nested fashion
```
{% set aboutPages = craft.entries.section('pages').all() %}
{% if aboutPages | length %}
    {% nav page in aboutPages %}
        <li>
            {% if page.parent %}[Parent: {{ page.parent.title }}] {{ page.getLink() }}
            {% ifchildren %}
                <ul>
                    {% children %}
                </ul>
            {% endifchildren %}
        </li>
    {% endnav %}
{% endif %}
```
3. That should output 

- Parent 1
- Parent 2
  - [Parent: Parent 2] Child 1
  - [Parent: Parent 2] Child 2

4. Now 'Disable' the root entry 'Parent 2'

### Expected behavior
I would expect that, since Parent 2 is disabled, it would not show and its children would not either. Something like:
- Parent 1

And if "Child 1" and "Child 2" were to show, I would think the `{{ page.parent.title }}` tag would show their actual parent

### Actual behavior
However, it seems to output the children of "Parent 2" anyway, and thinks their parent is "Parent 1"
- Parent 1
  - [Parent: Parent 1] Child 1
  - [Parent: Parent 1] Child 2

### Craft CMS version

3.7.27.2

### PHP version

7.3.33

### Database type and version

MySQL 5.7.37

**Comment by brandonkelly** [comment]
> I would expect that, since Parent 2 is disabled, it would not show and its children would not either.

Good call. Just fixed this for the next release!

**Comment by brandonkelly** [comment]
Just released 3.7.39 with this change.
